src/Phan/Plugin/Internal/BaselineLoadingPlugin.php

```diff
@@ -75,7 +85,32 @@ public function shouldSuppressIssue(
         array $parameters,
         ?Suggestion $suggestion
     ): bool {
-        return \in_array($issue_type, $this->file_suppressions[$context->getFile()] ?? [], true);
+        $suppressed_by_file = \in_array($issue_type, $this->file_suppressions[$context->getFile()] ?? [], true);
+        if($suppressed_by_file)
+            return true;
+
+        // Not suppressed by file, check for suppression by directory
+
+        $parts = explode('/', $context->getFile());
```

Comment:
This should be `str_replace('\\', '/', $context->getFile())` to match the normalization done on the directories this is compared against
```suggestion
        $parts = explode('/', str_replace('\\', '/', $context->getFile()));
```